sources/packages/api/src/services/education-program/education-program.service.ts

```diff
@@ -105,4 +106,39 @@ export class EducationProgramService extends RecordDataModelService<EducationPro
       return summaryItem;
     });
   }
+
+  /**
+   * Gets the program with respect to the programId
+   * @param programId Id of the Program.
+   * @returns summary for location
+   */
+  async getLocationPrograms(programId: number): Promise<EducationProgramModel> {
```

Comment:
In cases where we are returning almost all data from the entity and when the query will not be executed in a heavy way where the performance is critical, I would recommend using the out-of-box method from Typeorm to load the object using the Id.

Reply:
Discussed